pkg/types/currency_pair.go

```diff
@@ -59,21 +175,50 @@ func CurrencyPairString(base, quote string) string {
 	return cp.String()
 }
 
+// CurrencyPairFromString creates a currency pair from a string. Non-capitalized inputs are sanitized and the resulting
+// currency pair is validated.
 func CurrencyPairFromString(s string) (CurrencyPair, error) {
 	split := strings.Split(s, "/")
 	if len(split) != 2 {
-		return CurrencyPair{}, fmt.Errorf("incorrectly formatted CurrencyPair: %s", s)
+		return CurrencyPair{}, fmt.Errorf("incorrectly formatted CurrencyPair: %q", s)
+	}
+
+	base, err := sanitizeAssetString(split[0])
+	if err != nil {
+		return CurrencyPair{}, err
 	}
+
+	quote, err := sanitizeAssetString(split[1])
+	if err != nil {
+		return CurrencyPair{}, err
+	}
+
 	cp := CurrencyPair{
-		Base:  strings.ToUpper(split[0]),
-		Quote: strings.ToUpper(split[1]),
+		Base:  base,
+		Quote: quote,
 	}
 
 	return cp, cp.ValidateBasic()
 }
 
+func sanitizeAssetString(s string) (string, error) {
+	if IsLegacyAssetString(s) {
+		s = strings.ToUpper(s)
+	} else {
+		token, address, chainID, err := SplitDefiAssetString(s)
+		if err != nil {
+			return "", fmt.Errorf("incorrectly formatted asset: %q: %w", s, err)
+		}
+
+		token = strings.ToUpper(token)
+		s = strings.Join([]string{token, address, chainID}, fieldSeparator)
+	}
```

Comment:
we do we only upper the token?

Reply:
yes, wnat to preserve any formatting that might make addresses / chainIDs unique